core/src/test/java/com/coderanch/util/cli/InputUtilityTest.java

```diff
@@ -0,0 +1,126 @@
+package com.coderanch.util.cli;
+
+import junit.framework.TestCase;
+import org.junit.experimental.theories.Theories;
+import org.junit.experimental.theories.Theory;
+import org.junit.runner.RunWith;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+
+/**
+ * Tests the {@link InputUtility} class.
+ */
+@RunWith(Theories.class)
+public class InputUtilityTest {
+
+
+    /**
+     * @throws IOException
+     */
+    @Theory()
+    public void nextString_validatesCorrectInput() throws IOException {
+        var testCase = "input";
+        try(var s = new ByteArrayInputStream(testCase.getBytes())) {
+            var iu = new InputUtility(s);
+            var result =  iu.nextString("prompt", s1 -> s1.equals(testCase));
+            assertThat("Matches testCase", testCase, is(equalTo(result)));
+        }
+    }
+
+    /**
+     * @throws IOException
+     */
+    @Theory()
+    public void nextInt_validatesCorrectInput() throws IOException {
+        var testCase = "1";
+        try(var s = new ByteArrayInputStream(testCase.getBytes())) {
+            var iu = new InputUtility(s);
+            var result =  iu.nextInt("prompt", n -> n == 1);
```

Comment:
Don't use the predicate to make your assertion, that's what the `assertThat` call is for.